### PR TITLE
feat: update retry methods and queries on sql repository

### PR DESCRIPTION
**Pull request by joaoandre-avaiga**

### CODE REVIEW

File: taipy/core/_repository/_sql_repository.py [hunk 1]
```diff
@@ -73,19 +83,24 @@ def _exists(self, entity_id: str):
...
     def _load(self, entity_id: str) -> Entity:
-        query = self.table.select().filter_by(id=entity_id)
-
-        if entry := self.db.execute(str(query.compile(dialect=sqlite.dialect())), [entity_id]).fetchone():
-            entry = self.model_type.from_dict(entry)
-            return self.converter._model_to_entity(entry)
+        query = self.table.select().filter_by(id=entity_id).order_by(text("updated_at DESC"))
+
+        for _ in range(self.MAX_MODEL_NOT_FOUND_RETRY):
+            if entry := self.db.execute(str(query.compile(dialect=sqlite.dialect())), [entity_id]).fetchall():
```
Comment (jrobinAV) [inline taipy/core/_repository/_sql_repository.py:89]: The behavior we want is to have at most one element returned by this query.
We should at least log a warning if it is not the case.
To investigate why, maybe we can exit as well.

File: taipy/core/_repository/_sql_repository.py [hunk 2]
```diff
@@ -219,12 +235,17 @@ def __get_entities_by_config_and_owner(
     @_retry_repository_operation(__EXCEPTIONS_TO_RETRY)
     def __insert_model(self, model: ModelType):
         query = self.table.insert()
```
Comment (jrobinAV) [inline taipy/core/_repository/_sql_repository.py:237]: For investigation purposes, we should check first if an entity with the same id already exists.

Comment (joaoandre-avaiga) [reply taipy/core/_repository/_sql_repository.py:237]: We do check it on the `_save` function, before calling `__insert_model`. Do you mean check again on this function?

File: taipy/core/_repository/_sql_repository.py [hunk 3]
```diff
@@ -26,7 +29,14 @@
...
 class _SQLRepository(_AbstractRepository[ModelType, Entity]):
-    __EXCEPTIONS_TO_RETRY = (OperationalError,)
+    __EXCEPTIONS_TO_RETRY = (
+        OperationalError,
+        sqlite3.OperationalError,
+        sqlite3.InterfaceError,
+        IndexError,
+        StopIteration,
+    )
+    MAX_MODEL_NOT_FOUND_RETRY = 30
```
Comment (jrobinAV) [inline taipy/core/_repository/_sql_repository.py:39]: Isn't it too much? Or is it for debugging and investigation purpose?

**Review by jrobinAV — APPROVED** [review]

**Comment by joaoandre-avaiga** [comment]
@jrobinAV I think Im not gonna merge this PR while we discuss about removing sql.

**Comment by jrobinAV** [comment]
All right. Sounds fair.

**Comment by jrobinAV** [comment]
Since we decided to remove the sql option for repository type, should we close the PR and delete the branch ?

**Comment by joaoandre-avaiga** [comment]
Closing this since we decided to remove SQL Repository

**Pull request closed without merging**